conda_lock/conda_lock.py

```diff
@@ -36,6 +36,7 @@
 
 CONDA_PKGS_DIRS = None
 DEFAULT_PLATFORMS = ["osx-64", "linux-64", "win-64"]
+FILENAME_TOKENS = frozenset(["platform", "dev-dependencies", "spec-hash"])
```

Comment:
Perhaps expanding these to be a dict, e.g. 
```py
FILENAME_TOKENS = {
    "platform": "the conda-subdir",
    "dev-dependencies": "..."
    "spec-hash": "..."
}
```
And these can be included in the `help`.

Reply:
Turns out it's kind of tricky to have newlines in the help for an argument, and also to have a formatted string as your docstring, so I just added a hardcoded description of the tokens to the command help.